src/v/cluster/topic_table_probe.h

```diff
@@ -24,10 +24,23 @@ class topic_table_probe {
     void handle_topic_creation(create_topic_cmd::key_t);
     void handle_topic_deletion(const delete_topic_cmd::key_t&);
 
+    void handle_update_addition(const model::ntp& ntp);
+    void handle_update_removal(const model::ntp& ntp);
+
 private:
+    void setup_metrics();
+
+    void increment_moving_to_partitions() { moving_to_partitions += 1; }
+    void increment_moving_from_partitions() { moving_from_partitions += 1; }
+    void decrement_moving_to_partitions() { moving_to_partitions -= 1; }
+    void decrement_moving_from_partitions() { moving_from_partitions -= 1; }
+
     const topic_table& _topic_table;
     absl::flat_hash_map<model::topic_namespace, ss::metrics::metric_groups>
       _topics_metrics;
+    ss::metrics::metric_groups _metrics;
+    size_t moving_to_partitions = 0;
+    size_t moving_from_partitions = 0;
```

Comment:
maybe we should use signed integers here to avoid wild jumps if we slightly mess up arithmetic :)

Reply:
done